File: classic-cs-problems/1-fibonacci/4-iteration.py
def fibonacci(n: int) -> int:
    if n == 0: return n
    last: int = 0
    next: int = 1

    for _ in range(1, n):
        last, next = next, last + next
    return next


if __name__ == "__main__":
    print(fibonacci(42))

File: classic-cs-problems/1-fibonacci/4-iteration.js
const fibonacci = n => {
  if (n === 0) return n

  let last = 0
  let next = 1

  for (let i = 1; i < n; i++) {
    [last, next] = [next, last + next]
  }

  return next
}


console.log(fibonacci(42))
